### PR TITLE
[HOPSWORKS-1391] Extend Pandas and Numpy helper classes

### PR DESCRIPTION
**Pull request by jimdowling**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1391

### CODE REVIEW

File: hops/numpy_helper.py [hunk 1]
```diff
@@ -36,9 +125,65 @@ def save(hdfs_filename, data):
...
+
+def savez(hdfs_filename, *args, **kwds):
+    """
+    Save several arrays into a single file in uncompressed .npz format in HDFS
+    If arguments are passed in with no keywords, the corresponding variable names, in the .npz file, are ‘arr_0’, ‘arr_1’, etc. 
+    If keyword arguments are given, the corresponding variable names, in the .npz file will match the keyword names.
+
+    Args:
+       :hdfs_filename: You can specify either a full hdfs pathname or a relative one (relative to your Project's path in HDFS)
+       :args : Arguments, optional
```
Comment (moritzmeister) [inline hops/numpy_helper.py]: Might break the docs otherwise
```suggestion
       :args: Arguments, optional
```

File: hops/numpy_helper.py [hunk 2]
```diff
@@ -36,9 +125,65 @@ def save(hdfs_filename, data):
...
+               Arrays to save to the file. Since it is not possible for Python to know the names of the arrays outside savez, 
+               the arrays will be saved with names “arr_0”, “arr_1”, and so on. These arguments can be any expression.
+       :kwds : Keyword arguments, optional
```
Comment (moritzmeister) [inline hops/numpy_helper.py]:
```suggestion
       :kwds: Keyword arguments, optional
```

File: hops/numpy_helper.py [hunk 3]
```diff
@@ -36,9 +125,65 @@ def save(hdfs_filename, data):
...
+
+def savez_compressed(hdfs_filename, *args, **kwds):
+    """
+    Save several arrays into a single file in uncompressed .npz format in HDFS
+    If arguments are passed in with no keywords, the corresponding variable names, in the .npz file, are ‘arr_0’, ‘arr_1’, etc. 
+    If keyword arguments are given, the corresponding variable names, in the .npz file will match the keyword names.
+
+    Args:
+       :hdfs_filename: You can specify either a full hdfs pathname or a relative one (relative to your Project's path in HDFS)
+       :args : Arguments, optional
```
Comment (moritzmeister) [inline hops/numpy_helper.py]:
```suggestion
       :args: Arguments, optional
```

File: hops/numpy_helper.py [hunk 4]
```diff
@@ -36,9 +125,65 @@ def save(hdfs_filename, data):
...
+               Arrays to save to the file. Since it is not possible for Python to know the names of the arrays outside savez, 
+               the arrays will be saved with names “arr_0”, “arr_1”, and so on. These arguments can be any expression.
+       :kwds : Keyword arguments, optional
```
[the remark this answers is not in the document]
Comment (jimdowling) [reply hops/numpy_helper.py]: Ok, thanks. i fixed these.

**Comment by jimdowling** [comment]
I have tested this on a VM, added some tests to it_test notebook that work. I removed support for HDF as it requires the tables library (not included in our base env), so they developers can do it themselves if they want to use hdf.

**Pull request merged**